### ISSUE TITLE
inotify generating errors for failures on files that are treated as directories. 

### ISSUE THREAD

**Issue by kswagler-rh** [osquery/osquery#8240]
# Bug report

When running osquery with file events enabled on Linux systems an error is generated  when a file is attempted to be added as a directory. Eg.  
`
osqueryd[29801]: W1121 14:47:32.353986 29801 inotify.cpp:371] Could not add inotify watch on: /usr/bin/rpm/
`
`
$ ls -l /usr/bin/rpm
`
`
-rwxr-xr-x. 1 root root 20520 Nov 13 01:00 /usr/bin/rpm
`

### What operating system and version are you using?

Fedora 39

### What version of osquery are you using?

5.10.2

### What steps did you take to reproduce the issue?

Add a directory with files such as /usr/bin to FIM events
Run osquery in verbose mode

### What did you expect to see?

No warnings on adding a file listener to a file

### What did you see instead?

Warnings that a file could not have an inotify watch because it is not a directory.

**Comment by m8ram** [comment]
The warning comes from https://github.com/osquery/osquery/blob/df26c47f3598aa19a913b5ac53037b05e426c33a/osquery/events/linux/inotify.cpp#L370-L371

**Comment by dherder** [comment]
@kswagler-rh can you share your FIM config file please?

**Comment by kswagler-rh** [comment]
> @kswagler-rh can you share your FIM config file please?
@dherder  here's the config from Fleet 

```
      file_paths:
        tmp:
          - /tmp/%%
        binaries:
          - /usr/bin/%%
          - /usr/sbin/%%
          - /bin/%%
          - /sbin/%%
          - /usr/local/bin/%%
          - /usr/local/sbin/%%
          - /usr/lib/modules/%%
          - /usr/lib64/%%
        configuration:
          - /etc/passwd
          - /etc/shadow
          - /etc/ld.so.conf
          - /etc/ld.so.conf.d/%%
          - /etc/pam.d/%%
          - /etc/resolv.conf
          - /etc/rc%/%%
          - /etc/hosts
          - /etc/hostname
          - /etc/fstab
          - /etc/crontab
          - /etc/cron%/%%
```

**Comment by getvictor** [comment]
@kswagler-rh I can only reproduce this issue if there is a symlink we're trying to watch that's pointing at `/usr/bin/rpm`

Can you double check? If you create a file like `/usr/bin/file_no_symlink_knows_about` will you get a warning about it from osqueryd?

If not, then this is likely a duplicate of #8354

**Comment by directionless** [comment]
Fixed by: #8399
